frontend/isoko/src/pages/SearchResults.tsx

```diff
@@ -0,0 +1,404 @@
+      const dateComponents = props.time.split(',');
+      return dateComponents[0];
+   };
+
+   const displayTime = () => {
+      return moment(props.time, 'llll').format('h:mm A');
+   };
+
+   const filterByTime = (business, day) => {
+      // business doesn't have hours listed
+      if (Object.keys(business.hours).length === 0) {
+         return false;
+      }
+      const timestring = business.hours[day];
+
+      if (timestring === 'Closed') {
+         return false;
+      }
+
+      const splitstring = timestring.split(' - ');
+
+      const openString =
+         splitstring[0].substring(0, splitstring[0].length - 2) +
+         ' ' +
+         splitstring[0].slice(-2);
+
+      const closedString =
+         splitstring[1].substring(0, splitstring[1].length - 2) +
+         ' ' +
+         splitstring[1].slice(-2);
+      const open = moment(openString, 'h:mm a');
+      const closed = moment(closedString, 'h:mm a');
+
+      const now = moment(props.time, 'h:mm a');
+
+      if (now.isAfter(open) && now.isBefore(closed)) {
+         return true;
+      }
+
+      return false;
+   };
+
+   // when a user clicks on one of the minority tags on the left, it will be remove the tag from the search
+   const removeTag = (business, tag) => {
+      const tags = business.tags;
+      const selectedTags = [];
+
+      props.minorityTags.forEach((t, i) => {
+         selectedTags.push(t);
+      });
+
+      const tagIndex = selectedTags.indexOf(tag);
+      selectedTags.splice(tagIndex, 1);
+
+      // if the tag to remove isn't associated with this business, it should still be displayed
+      if (tags.indexOf(tag) === -1) {
+         return true;
+      }
+
+      // if the tag to remove is the only one associated with this business, it should be removed from results
+      if (tags.length === 1) {
+         return false;
+      }
+
+      // if there are more minorities to filter by, check if other tags associated with business fulfill them
+      if (selectedTags.length > 0) {
+         return selectedTags.some((t) => tags.includes(t));
+      }
+   };
+
+   // filter businesses based on sidebar components
+   const filterBusinesses = (e, key, businesses) => {
+      // filter by "open now"
+      if (key == 'time') {
+         const day = dayOfWeek();
+         console.log('day');
+         console.log(day);
+         console.log(e.target.checked);
+         // filter by time if box is checked, if unchecked, revert to original results
+         if (e.target.checked) {
+            setFilteredBusinesses(
+               businesses.filter((b) => filterByTime(b, day))
+            );
+         } else {
+            setFilteredBusinesses(businessResults);
+         }
+      }
+
+      // minority tag removed
+      else {
+         setFilteredBusinesses(businesses.filter((b) => removeTag(b, key)));
+         const tagIndex = props.minorityTags.indexOf(key);
+         props.minorityTags.splice(tagIndex, 1);
+      }
+   };
```

Comment:
I'm a bit confused about what this function does/when it is called. Is this called each time a user clicks the open now button and each time they remove a tag?

Reply:
yeah for some reason i thought it would be a good idea to put all the filtering functionality in one function and just pass a key to filter on, but these are split into separate functions in my latest commit